File: app/models/transaction.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum
from datetime import datetime
from bson import ObjectId

class PaymentMethod(str, Enum):
    BANK_TRANSFER = "Bank Transfer"
    CASH = "Cash"
    ONLINE = "Online"  # Paystack/Flutterwave

class TransactionStatus(str, Enum):
    PENDING = "Pending"
    VERIFIED = "Verified"
    REJECTED = "Rejected"

class TransactionBase(BaseModel):
    user_id: str
    contribution_id: Optional[str] = None  # Specific contribution being paid for (optional, could be bulk)
    amount: float
    payment_method: PaymentMethod
    reference_number: Optional[str] = None # Bank ref or Transaction ID
    proof_url: Optional[str] = None # URL to uploaded image
    status: TransactionStatus = TransactionStatus.PENDING
    remarks: Optional[str] = None

class TransactionCreate(TransactionBase):
    pass

class TransactionInDB(TransactionBase):
    id: Optional[str] = Field(alias="_id", default=None)
    verified_by: Optional[str] = None # Admin User ID
    created_at: datetime = Field(default_factory=datetime.utcnow)
    verified_at: Optional[datetime] = None

    class Config:
        populate_by_name = True
        arbitrary_types_allowed = True
        json_encoders = {ObjectId: str}

File: models/Transaction.js
const mongoose = require('mongoose');

const transactionSchema = new mongoose.Schema({
    user_id: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    amount: { type: Number, required: true },
    type: { type: String, enum: ['Contribution', 'Donation', 'Welfare', 'Project'], default: 'Contribution' },
    payment_method: { type: String, required: true }, // "Bank Transfer", "Cash", etc.
    reference_number: String,
    proof_url: String, // Path to receipt/pdf
    category: { type: String, enum: ['Contribution', 'Donation', 'Welfare', 'Project'], default: 'Contribution' },
    status: { type: String, enum: ['Pending', 'Verified', 'Rejected'], default: 'Pending' },
    contribution_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Contribution' }, // Optional link
    month: String, // Single month (deprecated, use months)
    months: [String], // Array of months: ["2026-01", "2026-02"]
    message: String, // Motivation / Note
    verified_by: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
    verified_at: Date,
    created_at: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Transaction', transactionSchema);
